### PR TITLE
PHP Data Structures

### PR DESCRIPTION
**Pull request by dguhl**
This PR proposes doc stubs for the PHP Data Structures extension. The signatures are compatible to PHP 7.1, like defined in your projects' `composer.json` and it features the splat operator in signatures for any indefinite number of parameters, as well as return types on functions. I included notes and warnings as separate paragraphs, as well as callback signatures as code blocks within the docs.

The stubs in this PR shall promote the usage of the new [PHP 7 Data Structure extension](https://www.php.net/manual/en/book.ds.php). Their usage also familiarize the developer to the essential usage principle of generic types and functions, as described in the draft [PHP RFC "Generic Types and Functions"](https://wiki.php.net/rfc/generics).

I look forward to your feedback.

### CODE REVIEW

**Comment by dguhl** [comment]
I suggest to squash the commits before merge, to preserve history's meaningfulness.

File: ds/ds.php [hunk 1]
```diff
@@ -0,0 +1 @@
+<?php
```
Comment (andrey-sokolov) [inline ds/ds.php:1]: Please, move code from ds_c.php to ds.php. There is no actual need in this empty file.

Comment (dguhl) [reply ds/ds.php:1]: Done. Also deleted duplicate classes which somehow came into the file. 🤔

**Comment by andrey-sokolov** [comment]
Thank you for this contribution.
> I suggest to squash the commits before the merge, to preserve history's meaningfulness.
Let's split the PR then in separate parts - 1) `Core_c.php` change 2) `.idea` folder related changes 3) actual Data structures extension.

**Comment by dguhl** [comment]
I rather would like to deminish the changes of Core_c.php and .idea, as changing them was not my intention. This PR should purely reflect the Data Structures extension.

Since the checks have failed, I will commit another change fixing it all along.

**Comment by andrey-sokolov** [comment]
@dguhl Hey, would you mind to resolve a merge conflict to allow this PR to be merged in master?

**Comment by dguhl** [comment]
I was not aware there was any. The checks all went through clear.

**Comment by dguhl** [comment]
All checks passed again. There is no merge conflict with master. 🤔

**Comment by ondrejmirtes** [comment]
FYI these stubs are missing some things, like `Ds\Map::get()` (https://www.php.net/manual/en/ds-map.get.php).

**Comment by ondrejmirtes** [comment]
See: https://github.com/phpstan/phpstan/issues/3470

**Comment by dguhl** [comment]
I'll look into it.

**Comment by dguhl** [comment]
I finished my checks and discovered multiple missing methods on Map as well as a few constructors. I created PR #833 to amend my contribution.

**Pull request merged**